source/common/network/connection_impl.cc

```diff
@@ -529,6 +533,12 @@ void ConnectionImpl::onWriteReady() {
     }
   }
 
+  // Disable the delayed close timer since data is still being flushed. The timer should only
+  // trigger after a delayedCloseTimeout() period of inactivity.
+  if (delayed_close_timer_ != nullptr) {
+    delayed_close_timer_->disableTimer();
```

Comment:
Is there any reason to disable the timer here? Won't we boost it below? (Mainly just wondering if we can simplify slightly but there might be a good reason to keep this disable here.)

Reply:
I had missed that libevent's `event_add()` would reset an already active timer with the newly provided timeout value. As you point out, this is not necessary and I have removed it.